docs-src/concepts/what-is-worker-versioning.md

```diff
@@ -0,0 +1,223 @@
+version, those Workers receive no Tasks.
+
+Now, imagine you need change the Workflow for some reason.
+
+Add `2.0` to the sets as the new default:
+
+| set 1         | set 2 (default) |
+| ------------- | --------------- |
+| 1.0 (default) | 2.0 (default)   |
+
+All new Workflows started on the Task Queue have their first tasks assigned to version `2.0`.
+Existing `1.0` Workflows keep generating tasks targeting `1.0`.
+Each deployment of Workers receives their respective Tasks.
+This same concept carries forward for each new incompatible version.
+
+Maybe you have a bug in `2.0`, and you want to make sure all open `2.0` Workflows switch to some new
+code as fast as possible. So, you'll add `2.1` to the sets, marking it as compatible with `2.0`. Now your
+sets look like this:
+
+| set 1         | set 2 (default) |
+| ------------- | --------------- |
+| 1.0 (default) | 2.0             |
+|               | 2.1 (default)   |
+
+All new Workflow Tasks that are generated for Workflows whose last Workflow Task completion was on
+version `2.0` are now assigned to version `2.1`. Because you specified that `2.1` is compatible with
+`2.0`, Temporal Server assumes that Workers with this version can process the existing Event
+Histories successfully.
+
+Continue with your normal development cycle, adding a `3.0` version.
+Nothing new here:
+
+| set 1         | set 2         | set 3 (default) |
+| ------------- | ------------- | --------------- |
+| 1.0 (default) | 2.0           | 3.0 (default)   |
+|               | 2.1 (default) |                 |
+
+Now imagine that version `3.0` doesn't have an explicit bug, but something about the business logic
+is less than ideal. You are okay with existing `3.0` Workflows running to completion, but you want new
+Workflows to use the old `2.x` branch. This operation is supported by performing an update targeting
+`2.1` (or `2.0`) and setting it's set as the current default, which results in these sets:
+
+| set 1         | set 3         | set 2 (default) |
+| ------------- | ------------- | --------------- |
+| 1.0 (default) | 3.0 (default) | 2.0             |
+|               |               | 2.1 (default)   |
+
+Now new Workflows start on `2.1`.
+
+#### Permitted and forbidden operations on version sets
+
+A request to change the sets can do one of the following:
+
+- Add a version to the sets as new the default version in a new overall-default compatible set.
+- Add a version to an existing set that's compatible with an existing version.
+  - Optionally making it the default for that set
+  - Optionally making that set the overall-default set.
+- Promote a version within an existing set to become the default for that set.
+- Promote a set to become the overall-default set.
+
+You can't explicitly delete versions. This helps you avoid the situation in which Workflows
+accidentally become stuck with no means of making progress because the version they're associated
+with no longer exists.
```

Comment:
```suggestion
You can't explicitly delete versions.This helps you avoid the situation in which Workflows accidentally become stuck with no means of making progress because the version they're associated with no longer exists.
```